### ISSUE TITLE
a logical bug when calculate 3-pixel error ?

### ISSUE THREAD

**Issue by chenchr** [JiaRenChang/PSMNet#83]
Hello. Thans for the code.
In ```finetune.py```, I found that there maybe a logical error to caculate the correct estimated pixel in line 131 : 
```
disp_true[index[0][:], index[1][:], index[2][:]] < 3)+(disp_true[index[0][:], index[1][:], index[2][:]] < true_disp[index[0][:], index[1][:], index[2][:]]*0.05)
```
here the code using adding ```+``` to conseder two condition. However, in torch the add opertation of ByteTensor seems will accumulate. I think here should use multiplication.

**Comment by chenchr** [comment]
Sorry. Using multiplication seems wrong too... Instead, maybe there shoud follows a line: 
``` correct[correct > 0] = 1 ```

**Comment by JiaRenChang** [comment]
@chenchr 
Hi, I think that `+` performs `or` operation to ByteTensor.
The criterion of 3px error is that `error<3 or error < true disp *0.05`.

**Comment by chenchr** [comment]
@JiaRenChang Yes, but test in pytorch when a pixel with both ```error<3``` and ```error < true_disp * 0.05```, the ```+```` will accumulate to 2, different from numpy.

**Comment by JiaRenChang** [comment]
@chenchr 
Hi, I tried pytorch 0.4.0 and 0.4.1, both module expressed same behavior.
![image](https://user-images.githubusercontent.com/11732099/45753646-10c9d800-bc4c-11e8-8d08-63a99cfa1b48.png)

**Comment by chenchr** [comment]
@JiaRenChang Maybe tensor A and B should have intersection on those equals one ?

**Comment by JiaRenChang** [comment]
@chenchr 
In our case, we need union on those equal one.

**Comment by chenchr** [comment]
@JiaRenChang  Maybe I have some misunderstanding...  What I am consider is the following:
Like the code in image
![image](https://user-images.githubusercontent.com/10046830/45769076-18e83e80-bc71-11e8-88d2-4f5a0eafe1b1.png)
Suppose ```mask_a``` means mask for those pixel with ```error<3```, and ```mask_b``` means mask for those pixel with ```error < true_disp * 0.05```, if directly adding two mask, the output will have value ```2```.
However, to compute the ratio of correct estimated pixel, we only need count those pixel with mask larger than 0, instead of accumulate the value of mask. In ```finetune.py```, the code computing the ratio goes like this:
```
return 1-(float(torch.sum(correct))/float(len(index[0])))
```
Therefore, I think we should insert a line ``` correct[correct > 0] = 1``` to avoid the condition that the value of mask is 2.

**Comment by JiaRenChang** [comment]
@chenchr 
Oh, I understood your concern.
I will modify the code from `+` to `|` .
Thank you!!

**Comment by chenchr** [comment]
@JiaRenChang  Don't mention it. Thanks you for the discussion.
